### PR TITLE
Deploy the api to modal

### PR DESCRIPTION
**Pull request by bugsz**
This pr majorly
1. Refactor the original FastAPI
2. Add deployment to modal in `modal_api_server`

Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed
- [ ] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [ ] Ready for code review

### CODE REVIEW

File: sotopia/ui/modal_api_server.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        "apt-get install -y curl gpg lsb-release",
+        # Add Redis Stack repository
+        "curl -fsSL https://packages.redis.io/gpg | gpg --dearmor -o /usr/share/keyrings/redis-archive-keyring.gpg",
+        "chmod 644 /usr/share/keyrings/redis-archive-keyring.gpg",
+        'echo "deb [signed-by=/usr/share/keyrings/redis-archive-keyring.gpg] https://packages.redis.io/deb $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/redis.list',
+        "apt-get update",
+        "apt-get install -y redis-stack-server",
+    )
+    .pip_install(
+        "pydantic>=2.5.0,<3.0.0",
```
Comment (ProKil) [inline sotopia/ui/modal_api_server.py:50]: Why not just install sotopia? This is not useful for maintaining dependencies.

**Review by ProKil — CHANGES_REQUESTED** [review]
it looks good to me, but please make sure that the pytest is passed.

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]
Added the modal deployment process to the doc?

**Comment by bugsz** [comment]
> it looks good to me, but please make sure that the pytest is passed.

@ProKil Current `SotopiaFastAPI` inherits FastAPI and there should not be any issue. I tested on my end and it works well.

**Pull request merged**